scripts/generate_dag.py

```diff
@@ -222,6 +240,52 @@ def copy_custom_callables_to_dot_dir(
         )
 
 
+def build_images(dataset_id: str, env: str):
+    parent_dir = DATASETS_PATH / dataset_id / "_images"
+    if not parent_dir.exists():
+        return
+
+    image_dirs = copy_image_files_to_dot_dir(
+        dataset_id, parent_dir, PROJECT_ROOT / f".{env}"
+    )
+    for image_dir in image_dirs:
+        build_and_push_image(dataset_id, image_dir)
+
+
+def copy_image_files_to_dot_dir(
+    dataset_id: str, parent_dir: pathlib.Path, env_dir: pathlib.Path
+) -> typing.List[pathlib.Path]:
+    target_dir = env_dir / "datasets" / dataset_id
+    target_dir.mkdir(parents=True, exist_ok=True)
+    subprocess.check_call(
+        ["cp", "-rf", str(parent_dir), str(target_dir)], cwd=PROJECT_ROOT
+    )
+
+    return list_subdirs(target_dir / "_images")
+
+
+def build_and_push_image(dataset_id: str, image_dir: pathlib.Path):
+    image_name = f"{dataset_id}__{image_dir.name}"
```

Comment:
Is there a reason there are two underscores here?

Reply:
It's a way to visually separate the dataset from the container name, when either of them can contain underscores. Like this one:

![Screen Shot 2021-05-17 at 6 41 55 PM](https://user-images.githubusercontent.com/1208372/118565123-9b39a100-b73f-11eb-9dbb-3ebbd84ae952.png)

A hyphen works too. because container names and datasets always use snakecase. Let me know which you prefer.

Reply:
Got it - I just wanted to make sure it was intentional. I'm not sure which I like better - I think it's a little tough to see the difference between one underscore and two, but I think that adding a hyphen would possibly look like a mistake, so let's stick with the underscores, unless there's a hidden third option that is magically better than these two 🧙🏻

Reply:
That's a good consideration. Let's go with "dunders" then, it also keeps the snake case rule intact even for generated names.